core/domain/user_jobs_continuous.py

```diff
@@ -286,7 +293,105 @@ def _get_batch_job_manager_class(cls):
 
     @classmethod
     def _handle_incoming_event(cls, active_realtime_layer, event_type, *args):
-        pass
+        exp_id = args[0]
+
+        def _refresh_average_ratings(user_id, rating, old_rating):
+            realtime_class = cls._get_realtime_datastore_class()
+            realtime_model_id = realtime_class.get_realtime_id(
+                active_realtime_layer, user_id)
+
+            model = realtime_class.get(realtime_model_id, strict=False)
+            if model is None:
+                realtime_class(
+                    id=realtime_model_id, average_ratings=rating,
+                    num_ratings=1, realtime_layer=active_realtime_layer).put()
+            else:
+                num_ratings = model.num_ratings
+                average_ratings = model.average_ratings
+                num_ratings += 1
+                if average_ratings is not None:
+                    ratings_value = (
+                        average_ratings * (num_ratings - 1) + rating)
+                    if old_rating is not None:
+                        ratings_value -= old_rating
+                        num_ratings -= 1
+                    model.average_ratings = (ratings_value/num_ratings * 1.0)
```

Comment:
Is the 1.0 to make it a float? If so, I don't think it works: try doing "7/4 * 1.0" in python.